<launch>

  <arg name="gui" default="true"/>
  <arg name="downsample" default="true" />
  <arg name="input_cloud" default="/camera/depth_registered/points" />
  <arg name="input_indices" default="/fg_node/output/cluster_indices" />
  <arg name="input_classes" default="/fg_node/output/class" />
  <arg name="manager" default="/camera/realsense2_camera_manager" />

  <node name="fg_euclidean_clustering"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/EuclideanClustering $(arg manager)"
        clear_params="true">
    <remap from="~input" to="$(arg input_cloud)"/>
    <remap from="~input/cluster_indices" to="$(arg input_indices)"/>
    <rosparam>
      multi: true
      tolerance: 0.02
      min_size: 0
      max_size: 307200
      downsample_enable: true
      approximate_sync: true
      queue_size: 100
      cluster_filter: 1
      use_cluster_indices_header: true
    </rosparam>
  </node>

  <node name="fg_cluster_point_indices_decomposer"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/ClusterPointIndicesDecomposer $(arg manager)"
        clear_params="true">
    <remap from="~input" to="$(arg input_cloud)"/>
    <remap from="~target" to="fg_euclidean_clustering/output"/>
    <remap from="~boxes" to="fg_node/output/boxes"/>
    <remap from="~centroid_pose_array" to="fg_node/output/centroid"/>
    <rosparam>
      align_boxes: false
      align_boxes_with_plane: false
      force_to_flip_z_axis: false
      use_pca: true
      approximate_sync: true
      queue_size: 100
      use_cluster_indices_header: true
    </rosparam>
  </node>

  <node name="color_class_visualizer"
        pkg="jsk_rviz_plugins" type="classification_result_visualizer.py" >
    <remap from="~input/boxes" to="fg_node/output/boxes"/>
    <remap from="~input/classes" to="$(arg input_classes)" />
    <rosparam>
      text_offset_x: 0.0
      text_offset_y: -0.1
      text_offset_z: 0.0
      text_size: 0.07
      marker_lifetime: 0.5
    </rosparam>
  </node>

</launch>
